### PR TITLE
Makefile tweak

### PR DESCRIPTION
**Pull request by kfu02**
## Description
**Ignore everything in docs/build/. These files were autogenerated and do not need to be reviewed.**

Adds helpful shortcuts to our root Makefile for a better development experience. (Note: many binds Ubuntu-only, like our repo as a whole.) I would appreciate comments on both the function and the readability of the docs describing them.

Summary list:
 * `make run-*` - runs the specified version of code, e.g. `make run-sim` or `make run-soccer` (sim2play is still broken on this branch)
 * `make docs` - builds docs statically (local copy, not hosted on the internet)
 * `make open-docs` - opens statically built docs
 * `source ./source.bash` or `. ./source.bash` - sources ros2 and install in one command
 * `make again` - if repo built already with another method, builds faster by skipping CMake files

See Steps to Test for more detail.

### Test Case 1
1. Run `make docs`.
2. Run `make open-docs`.
3. Click on the "Getting Started" link of the webpage that opens up, and read through and try to use the new binds.

Expected result: Everything works as described in the documentation.

### CODE REVIEW

File: docs/source/installation/index.rst [hunk 1]
```diff
@@ -12,12 +12,12 @@ Clone the repository:
...
    git clone git://github.com/RoboJackets/robocup-software
 
-Run the setup script:
+Run the setup script to install all required dependencies:
 
 .. code-block:: sh
 
     cd robocup-software
-    util/ubuntu-setup
+    ./util/ubuntu-setup
```
Comment (kfu02) [inline docs/source/installation/index.rst:23]: also, run 
```
git submodule init
git submodule update
```

File: docs/source/getting_started/index.rst [hunk 2]
```diff
@@ -1,66 +1,192 @@
 Getting Started
```
Comment (kfu02) [inline docs/source/getting_started/index.rst:1]: add setup steps for grSim = 6 robots?

**Review by kfu02 — COMMENTED** [review]
TODOs for me

**Review by HussainGynai — APPROVED** [review]
LGTM, noticeable difference in build time.

**Review by kasohrab — APPROVED** [review]
lgtm

**Review by sseye6 — APPROVED** [review]
LGTM

**Comment by kfu02** [comment]
> LGTM, noticeable difference in build time.

love writing down things @kylestach uses and passing it off as my own work

**Pull request merged**